### PR TITLE
Support Envoy Proxy pod & container securityContext configuration.

### PR DESCRIPTION
**Pull request by qicz**
support  #1211 

```yaml
envoyDeployment:
  replicas: 2
  container:
    resources:
      requests:
        cpu: 100m
        memory: 512Mi
    securityContext:
      runAsUser: 2000
      allowPrivilegeEscalation: false
  pod:
    annotations:
      key1: val1
      key2: val2
    securityContext:
      runAsUser: 1000
      runAsGroup: 3000
      fsGroup: 2000
      fsGroupChangePolicy: "OnRootMismatch"
```

### CODE REVIEW

File: api/config/v1alpha1/shared_types.go [hunk 1]
```diff
@@ -40,23 +40,53 @@ const (
...
+
+	// TODO: Expose config as use cases are better understood, e.g. labels.
+}
+
+// KubernetesPodSpec defines the desired state of the Kubernetes pod resource.
+type KubernetesPodSpec struct {
 	// Replicas is the number of desired pods. Defaults to 1.
 	//
 	// +optional
 	Replicas *int32 `json:"replicas,omitempty"`
```
Comment (arkodg) [inline api/config/v1alpha1/shared_types.go:61]: Replicas should be part of `KubernetesDeploymentSpec`

Comment (qicz) [reply api/config/v1alpha1/shared_types.go]: did it

**Comment by qicz** [comment]
@arkodg pls review again

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**